pkg/workloads/cortex/client/cortex/binary/__init__.py

```diff
@@ -67,26 +67,28 @@ def run_cli(
 
     output = ""
     result = ""
-    result_found = False
+    processing_result = False
 
     for c in iter(lambda: process.stdout.read(1), ""):
         output += c
         if mixed_output:
-            if output[-2:] == "\n~":
-                result_found = True
-                result = "~"
+            if output[-2:] == "\n~" or output == "~":
+                processing_result = True
                 output = output[:-1]
-            if result_found:
-                output = output[:-1]
-                if c == "\n":
-                    result_found = False
+            if processing_result:
+                result += c
+                if (
+                    result[: len(MIXED_CORTEX_MARKER)] == MIXED_CORTEX_MARKER
+                    and result[-len(MIXED_CORTEX_MARKER) :] == MIXED_CORTEX_MARKER
+                    and len(result) > len(MIXED_CORTEX_MARKER)
```

Comment:
Does this address the extra new lines that are printed? We can discuss it live too